group-coordinator/src/main/java/org/apache/kafka/coordinator/group/streams/StreamsGroup.java

```diff
@@ -1049,4 +1057,21 @@ public StreamsGroupDescribeResponseData.DescribedGroup asDescribedGroup(
         return describedGroup;
     }
 
+    public void maybeSetShutdownRequested(final String memberId, final boolean shutdownApplication) {
```

Comment:
should we just have a setShutdownRequested method and do the if check on the calller's side?

Reply:
Good idea. Done